src/storage-types/src/connections/aws.rs

```diff
@@ -23,52 +23,132 @@ include!(concat!(
     env!("OUT_DIR"),
     "/mz_storage_types.connections.aws.rs"
 ));
+/// The materialize principal details
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
+pub struct MaterializePrincipal {
+    /// The role which Materialize will assume first to eventually
+    /// assume the customer's role via role chaining.
+    pub arn: String,
+    /// The external ID prefix, will need the connection ID as a suffix
+    /// to get the complete External ID to assume customer's role.
+    pub external_id_prefix: String,
+}
```

Comment:
As mentioned below, I think this type should disappear, with fields moving to the `ConnectionContext`...